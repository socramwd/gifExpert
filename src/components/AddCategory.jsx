
import { useState } from "react"
import PropTypes from 'prop-types'

// eslint-disable-next-line react/prop-types
export const AddCategory = ({onNewCategory}) => {
    const[inputValue, setInputValue] = useState('One Punch')

    const onInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const onSubmit = (e) => {
        e.preventDefault()

        const newInputValue = inputValue.trim()

        if( newInputValue.length <= 1) return

        onNewCategory(newInputValue)
        
        setInputValue('')
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}