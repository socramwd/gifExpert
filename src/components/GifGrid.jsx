
import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs"

// eslint-disable-next-line react/prop-types
export const GifGrid = ({category}) => {
    const { images, isLoading } = useFetchGifs(category)

    return (
        <>
            <section className="contenido">
                <h3 className="category_name">{category}</h3>

                {
                    isLoading && (
                        <p>...cargando...</p>
                    )
                }

                <div className="card-grid">
                    {
                        images.map( image => (
                            <GifItem
                                key={image.id}
                                { ...image } // Esparcimos todas las props asÃ­ no tenemos que poner tantas variables
                            />
                        ))
                    }
                </div>
            </section>
        </>
    )
}
