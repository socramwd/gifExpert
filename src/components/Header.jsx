
import { AddCategory } from "./AddCategory"

export const Header = ({onAddCategory}) => {
    return (
        <header className="header">
            <h1>GifExpertApp</h1>

            <AddCategory
                // setCategories={setCategories}
                onNewCategory={onAddCategory}
            />
        </header>
    )
}