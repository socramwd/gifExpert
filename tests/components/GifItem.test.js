import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem/>', () => {
    const title = 'socramdem'
    const url = 'https://one-punch.com/socramdev.jpg'

    test('Debe de hacer match con el snapshot', () => {
        const {container} = render(<GifItem title={ title } url={ url } />)
        expect( container ).toMatchSnapshot()
    })

    test('Debe de mostrar la imagen con el url y el alt indicado', () => {
        render(<GifItem title={ title } url={ url } />)
        
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('Debe de mostrar el tÃ­tulo del componente', () => {
        render(<GifItem title={ title } url={ url } />)

        expect( screen.getByText( title ) ).toBeTruthy()
    })
})