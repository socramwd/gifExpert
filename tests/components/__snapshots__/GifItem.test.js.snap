// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="socramdem"
      src="https://one-punch.com/socramdev.jpg"
    />
    <p>
      socramdem
    </p>
  </div>
</div>
`;
